src/openms/include/OpenMS/MATH/STATISTICS/StatisticFunctions.h

```diff
@@ -72,108 +70,110 @@ namespace OpenMS
+      }
     }
-  }
-  /**
-  @brief Helper function checking if an iterator and a co-iterator both have a next element
-
-  @exception Exception::InvalidRange is thrown if the iterator do not end simultaneously
 
-  @ingroup MathFunctionsStatistics
-  */
-  template <typename IteratorType1, typename IteratorType2>
-  static void checkIteratorsAreValid(
+    /**
+       @brief Helper function checking if an iterator and a co-iterator both have a next element
+       
+       @exception Exception::InvalidRange is thrown if the iterator do not end simultaneously
+       
+       @ingroup MathFunctionsStatistics
+    */
+    template <typename IteratorType1, typename IteratorType2>
+    static void checkIteratorsAreValid(
       IteratorType1 begin_b, IteratorType1 end_b,
       IteratorType2 begin_a, IteratorType2 end_a)
-  {
-    if(begin_b != end_b && begin_a == end_a)
     {
-      throw Exception::InvalidRange(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION);
+      if (begin_b != end_b && begin_a == end_a)
+      {
+        throw Exception::InvalidRange(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION);
+      }
     }
-  }
-  /**
-    @brief Calculates the sum of a range of values
-
-    @ingroup MathFunctionsStatistics
-  */
-  template <typename IteratorType>
-  static double sum(IteratorType begin, IteratorType end)
-  {
-    return std::accumulate(begin, end, 0.0);
-  }
-
-  /**
-    @brief Calculates the mean of a range of values
-
-    @exception Exception::InvalidRange is thrown if the range is NULL
-
-    @ingroup MathFunctionsStatistics
-  */
-  template <typename IteratorType>
-  static double mean(IteratorType begin, IteratorType end)
-  {
-    checkIteratorsNotNULL(begin, end);
-    return sum(begin, end) / std::distance(begin, end);
-  }
-
-  /**
-    @brief Calculates the median of a range of values
-
-    @param begin Start of range
-    @param end End of range (past-the-end iterator)
-    @param sorted Is the range already sorted? If not, it will be sorted.
-    @return Median (as floating point, since we need to support average of middle values)
-    @exception Exception::InvalidRange is thrown if the range is NULL
 
-    @ingroup MathFunctionsStatistics
-  */
-  template <typename IteratorType>
-  static double median(IteratorType begin, IteratorType end, bool sorted = false)
-  {
-    checkIteratorsNotNULL(begin, end);
-    if (!sorted)
+    /**
+       @brief Calculates the sum of a range of values
+       
+       @ingroup MathFunctionsStatistics
+    */
+    template <typename IteratorType>
+    static double sum(IteratorType begin, IteratorType end)
     {
-      std::sort(begin, end);
+      return std::accumulate(begin, end, 0.0);
     }
 
-    Size size = std::distance(begin, end);
-    if (size % 2 == 0)        // even size => average two middle values
+    /**
+       @brief Calculates the mean of a range of values
+      
+       @exception Exception::InvalidRange is thrown if the range is NULL
```

Comment:
A range is not NULL. A range is empty

Reply:
`StatisticFunctions` changes are not from me. That's why I removed it again from this PR.